File: javascript/src/oddwords.js
String.prototype.oddwords = function() {
    var words = this.split(' ');
    var result = [];
    
    for (var i = 0; i < words.length; i++) {
        var odd = (i % 2 != 0);
        var word = words[i];
        
        if (odd) {
            result.push(word.split('').reverse().join(''));
        } else {
            result.push(word);
        }
    }  
    
    return result.join(' ');
};

File: rebol/oddwords.r
REBOL [Title: "Odd Words"]

oddwords: func [phrase] [
    text: copy phrase
    words: parse text none
    
    for i 2 length? words 2 [
        word: pick words i
        reverse/part (find/last text word) (length? word)
    ]
    
    text
]

print oddwords "foo"
print oddwords "this is a simple phrase"
print oddwords "rebol is so cool"
